libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1693,6 +1693,8 @@ string YulUtilFunctions::copyValueArrayStorageToStorageFunction(ArrayType const&
 	solAssert(_fromType.dataStoredIn(DataLocation::Storage) && _toType.baseType()->isValueType(), "");
 	solAssert(_toType.dataStoredIn(DataLocation::Storage), "");
 
+	solUnimplementedAssert(_fromType.storageStride() == _toType.storageStride(), "");
```

Comment:
Why does this not throw for the modified test?

Reply:
Arrays have same base type (`uint256`). This function doesn't work for copying `byte[10]` to `bytes2[32]` for example

Reply:
Ah it's the base type, right

Reply:
Is this really an "UnimplementedAssert" or rather a regular "assert"?